core-library/component-styling/indicator/_indicator.scss

```diff
@@ -0,0 +1,262 @@
+@use "../../util/color" as color;
+@use "../../util/theme" as theme;
+@use "../../util/device" as device;
+@use "../../util/size" as size;
+@use "../../tokens/template-const" as template-const;
+@use "../../tokens/sizes" as token-size;
+@use "../../typography/typography" as typography;
+@use "../../tokens/text" as text;
+@use "../../tokens/partial/focus" as focus;
+@use "sass:map";
+
+@mixin selector {
+  lib-indicator#{template-const.$escape} {
+    @content;
+  }
+}
+
+@mixin create {
+  @include selector() {
+    @include layout();
+  }
+}
+
+@mixin size-sm {
+  .text {
+    padding: 2px 8px;
+    border-radius: 11px;
+
+    @include typography.fontSize(map.get(text.$body, level), 4);
+
+    &.rounded {
+      min-width: 8px;
+    }
+
+    > svg {
+      width: 12px;
+    }
+  }
+
+  .number {
+    padding: 6px 9px;
+
+    > span {
+      min-width: 14px;
+    }
+
+    @include typography.fontSize(map.get(text.$body, level), 4);
+
+    &.num-lg {
+      padding: 2px 8px;
+      border-radius: 11px;
+    }
+  }
+
+  .dot {
+    width: 6px;
+    height: 6px;
+  }
+}
+
+@mixin size-lg {
+  .text {
+    padding: 4px 8px;
+    border-radius: 15px;
+
+    @include typography.fontSize(map.get(text.$body, level), 2);
+
+    &.rounded {
+      padding: 2px 8px;
+      min-width: 12px;
+    }
+
+    > svg {
+      width: 16px;
+    }
+  }
+
+  .number {
+    padding: 7px 10px;
+
+    > span {
+      min-width: 18px;
+    }
+
+    @include typography.fontSize(map.get(text.$body, level), 2);
+
+    &.num-lg {
+      padding: 4px 8px;
+      border-radius: 15px;
+    }
+  }
+
+  .dot {
+    width: 10px;
+    height: 10px;
+  }
+}
+
+@mixin status-color {
+  &.info,
+  &.teal {
+    color: var(--information-text);
+  }
+
+  &.warning,
+  &.orange {
+    color: var(--warning-text);
+  }
+
+  &.critical,
+  &.red {
+    color: var(--critical-text);
+  }
+
+  &.generic,
+  &.grey {
+    color: color.mix-token(neutral, 900);
```

Comment:
these mix-tokens will need to moved to a fresh token sheet - if we don't it'll cause issues when we're trying to integrate dark mode.